src/main/java/org/radarbase/management/web/rest/ProjectResource.java

```diff
@@ -261,7 +262,12 @@ public ResponseEntity getAllSourcesForProject(@ApiParam Pageable pageable,
                     Boolean minimized) throws NotAuthorizedException {
         log.debug("REST request to get all Sources");
         ProjectDTO projectDto = projectService.findOneByName(projectName);
-        checkPermissionOnProject(getJWT(servletRequest), SOURCE_READ, projectDto.getProjectName());
+        RadarToken jwt = getJWT(servletRequest);
+        checkPermissionOnProject(jwt, SOURCE_READ, projectDto.getProjectName());
+        if (jwt.hasAuthority(PARTICIPANT)) {
+            throw new NotAuthorizedException(String.format("Client %s does not have "
```

Comment:
Would it be better to rephrase the mesage that a PARTICIPANT cannot perform this operation. We are not checking for permission here?